Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Themes/ThemeStateProperty.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    /// <summary>
+    /// Class to store information about a Theme property that contains values per available state
+    /// </summary>
+    [Serializable]
+    public class ThemeStateProperty
+    {
+        // TODO: Troy Add comments
+        public string Name;
+        public ThemePropertyTypes Type;
+        public List<ThemePropertyValue> Values;
+        public ThemePropertyValue StartValue;
+        public ThemePropertyValue Default;
+
+        public Shader TargetShader;
+        public string ShaderPropertyName;
+
+        [System.NonSerialized]
+        protected int ShaderPropertyID = -1;
+
+        private static readonly ThemePropertyTypes[] ShaderTypes =
+            { ThemePropertyTypes.Color, ThemePropertyTypes .ShaderFloat, ThemePropertyTypes.ShaderRange};
+
+        public static bool IsShaderPropertyType(ThemePropertyTypes type)
+        {
+            return ShaderTypes.Contains(type);
+        }
+
+        /// <summary>
+        /// Lazy loads shader property ID from Unity for the ShaderPropertyName
+        /// </summary>
+        /// <returns>integer key for current shader property to get/set shader values</returns>
+        public int GetShaderPropertyId()
+        {
+            if (ShaderPropertyID == -1)
+            {
+                ShaderPropertyID = Shader.PropertyToID(ShaderPropertyName);
+            }
+
+            return ShaderPropertyID;
+        }
+
+        // Properties below are outdated. They remain only for backward compatibility and migration purposes
+        [SerializeField]
+        private int PropId = -1; // i.e OptionIndex
```

Comment:
Isn't there a way to mark these as obsolete? https://answers.unity.com/questions/559529/how-to-mark-a-method-obsolete.html

Reply:
Yes there is....but I still need to access them to do the migration. And if I reference them Unity will throw errors:
![image](https://user-images.githubusercontent.com/25975362/64204632-10215600-ce4b-11e9-9f74-9ddf971afbd2.png)

Hmmm, I'm sure there must be a way to disable this obsolete error for the few code references I use it. I'll look into it